packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -208,17 +238,43 @@ function OrgSelect({
       return search.test(datum.orgName);
     });
 
-    return filtered;
+    setFilteredData(filtered);
   };
 
-  const renderedData = onChangeProp ? consumerFilteredData : filterData();
+  const didMount = useRef(false);
+  useEffect(() => {
+    // allow the user to provide filtered data
+    if (onChangeProp) {
+      return;
+    }
+
+    // manage serverside filtering for a global user
+    if (admin) {
+      // don't fetch data initially, only when filtering
+      if (!didMount.current) {
+        didMount.current = true;
+        return;
+      }
+
+      filterDataAsAdmin();
+      return;
+    }
+
+    // otherwise, filter the provided data clientside
+    filterData();
+  }, [value]);
+
+  useEffect(() => {
+    setFilteredData(data);
+  }, [data]);
```

Comment:
So...I _think_ we can get rid of this if we assume that if `value === ''` we show the initial data. Anything that's not an empty string we show the filtered data.

Reply:
I still needed this, but it allowed me to keep the logic about `value === ''` simple as a result.